core/chains/evm/client/pool.go

```diff
@@ -141,11 +142,17 @@ func (p *Pool) Dial(ctx context.Context) error {
 	})
 }
 
-// nLiveNodes returns the number of currently alive nodes
-func (p *Pool) nLiveNodes() (nLiveNodes int) {
+// nLiveNodes returns the number of currently alive nodes, as well as the highest block number and greatest total difficulty.
+func (p *Pool) nLiveNodes() (nLiveNodes int, blockNumber int64, totalDifficulty *utils.Big) {
 	for _, n := range p.nodes {
-		if n.State() == NodeStateAlive {
+		if s, num, td := n.StateAndLatest(); s == NodeStateAlive {
```

Comment:
For some reason every time I read this it makes me think of [chase and status](https://www.youtube.com/watch?v=hkDGGsuVKVo)